test/interactiveutil.jl

```diff
@@ -0,0 +1,110 @@
+### Testing download(url,[localfile],[silent=false])
+
+"Runs an expression, and returns stdout, stderr, and the result."
+function capture_output(expression)
+	originalSTDOUT = STDOUT
+	originalSTDERR = STDERR
+	(errRead, errWrite) = redirect_stderr()
+	(outRead, outWrite) = redirect_stdout()
+
+	result = eval(expression)	
+	# Ensure both streams are written to or readavailable waits forever.
+	print("\n")
+	write(STDERR, "\n")
+
+	close(errWrite)
+	close(outWrite)
+	errors = readavailable(errRead)
+	output = readavailable(outRead)
+	close(outRead)
+	close(errRead)
+	redirect_stderr(originalSTDERR)
+	redirect_stdout(originalSTDOUT)
+
+	errstr = join(map(x->Char(x), errors))
+	outstr = join(map(x->Char(x), output))	
+	# Remove the manually added \n from stdout and stderr
+	return (chop(outstr), chop(errstr), result)
+end
+
+port = 32744 #TODO - find generic way to get an unused port.
+html = "<html>success</html>"
+html_len = length(html)
+server_running = true
+response = """
+HTTP/1.1 200 OK
+Content-Length: $html_len
+Content-Type: text/html
+Connection: Closed
+
+$html"""
+
+function start_server()
+	server_running = true
+	@async begin
+		server = listen(port)
+		while server_running
+			sock = accept(server)
+			readline(sock)
+			write(sock, response)
+			close(sock)
+		end
+	end
+end
+
+function stop_server()
+	server_running = false
+end
+
+downloader = :unknown
+for checkcmd in (:curl, :wget, :fetch)
+	if success(pipeline(`which $checkcmd`, DevNull))
```

Comment:
@tkelman that points out another issue. On Windows `download` uses a `ccall` which means we shouldn't be performing this check on Windows at all. Prepending `@unix_only` to this for loop should solve the issue.